Lib/enum.py

```diff
@@ -789,27 +791,23 @@ def __iter__(self):
 
     def __repr__(self):
         cls = self.__class__
-        if self._name_ is not None:
-            return '<%s.%s: %r>' % (cls.__name__, self._name_, self._value_)
-        members, uncovered = _decompose(cls, self._value_)
-        return '<%s.%s: %r>' % (
-                cls.__name__,
-                '|'.join([str(m._name_ or m._value_) for m in members]),
-                self._value_,
-                )
+        if self._name_ is None:
+            members, uncovered = _decompose(cls, self._value_)
+            cls_name = self.__class__.__name__
+            self._name_ = '|'.join(str(m._name_ or m._value_) for m in members)
+        return '|'.join('%s.%s' % (cls.__name__, name) for name in self._name_.split('|'))
 
     def __str__(self):
         cls = self.__class__
-        if self._name_ is not None:
-            return '%s.%s' % (cls.__name__, self._name_)
-        members, uncovered = _decompose(cls, self._value_)
-        if len(members) == 1 and members[0]._name_ is None:
-            return '%s.%r' % (cls.__name__, members[0]._value_)
-        else:
-            return '%s.%s' % (
-                    cls.__name__,
-                    '|'.join([str(m._name_ or m._value_) for m in members]),
-                    )
+        if self._name_ is None:
+            members, uncovered = _decompose(cls, self._value_)
+            # if len(members) == 1 and members[0]._name_ is None:
+            #     self._name_ =  '%s' % (members[0]._value_, )
+            # else:
+            self._name_ = '%s' % '|'.join([
+                    str(m._name_ or m._value_) for m in members
+                    ])
+        return self._name_
```

Comment:
Now whichever of `__str__` and `__repr__` is called first will initialize `_name_` for all time. Even if they do the same thing now, they might drift in the future.  Should this initialization code be in a common function to make sure it's consistent?    One option would be to override name() and do it there.

Reply:
The name is now initialized when the pseudo-member is created.